x-pack/plugins/security_solution/scripts/endpoint/common/fleet_services.ts

```diff
@@ -209,9 +215,70 @@ export const waitForHostToEnroll = async (
     );
   }
 
-  log.debug(`Host [${hostname}] has been enrolled with fleet`);
+  log.info(`Host [${hostname}] has been enrolled with fleet`);
   log.verbose(found);
 
+  if (!awaitMetadata) {
+    return found;
+  }
+  log.info('Querying indices');
+
+  log.debug(
+    JSON.stringify(
+      await esClient?.search({
+        index: [
+          '.fleet-agents',
+          'metrics-endpoint.metadata_current_default',
+          '.metrics-endpoint.metadata_united_default',
+        ],
+      }),
```

Comment:
And actually - now that I'm looking at this code here - what is it even trying to do?  Just check that **any** data is in the index?  Can you explain what this `.search()` is attempting accomplish since you are not even capturing the result.

Reply:
so I believe this is trying to bypass the issue we were seeing with united metadata sometimes being unable to find docs in `.fleet-agents` even though they exist (technically, only searching `.fleet-agents` is necessary for this purpose). see https://github.com/elastic/kibana/blob/9a5dd789e071bea1c275c275b73ebbeb5c401b19/x-pack/test/security_solution_endpoint/services/endpoint.ts#L205. I'm still trying to work out the root cause. my current hypothesis is that the index's background refresh is going idle due to some changes in stateless ES.